package.json

```diff
@@ -1,36 +1,89 @@
 {
   "name": "@hookform/resolvers",
+  "amdName": "hookformResolvers",
   "version": "1.3.1",
-  "description": "React Hook Form validation resolvers: Yup, Joi, Superstruct, Zod and etc.",
-  "umd:main": "umd/index.js",
-  "unpkg": "umd/index.js",
-  "jsdelivr": "umd/index.js",
-  "sideEffects": false,
+  "description": "React Hook Form validation resolvers: Yup, Joi, Superstruct, Zod, Vest and etc.",
+  "main": "dist/resolvers.js",
+  "module": "dist/resolvers.module.js",
+  "umd:main": "dist/resolvers.umd.js",
+  "source": "src/index.ts",
+  "types": "dist/index.d.ts",
+  "exports": {
+    ".": {
+      "browser": "./dist/resolvers.module.js",
+      "umd": "./dist/resolvers.umd.js",
+      "import": "./dist/resolvers.mjs",
+      "require": "./dist/resolvers.js"
+    },
+    "./zod": {
+      "browser": "./zod/dist/zod.module.js",
+      "umd": "./zod/dist/zod.umd.js",
+      "import": "./zod/dist/zod.mjs",
+      "require": "./zod/dist/zod.js"
+    },
+    "./yup": {
+      "browser": "./yup/dist/yup.module.js",
+      "umd": "./yup/dist/yup.umd.js",
+      "import": "./yup/dist/yup.mjs",
+      "require": "./yup/dist/yup.js"
+    },
+    "./joi": {
+      "browser": "./joi/dist/joi.module.js",
+      "umd": "./joi/dist/joi.umd.js",
+      "import": "./joi/dist/joi.mjs",
+      "require": "./joi/dist/joi.js"
+    },
+    "./vest": {
+      "browser": "./vest/dist/vest.module.js",
+      "umd": "./vest/dist/vest.umd.js",
+      "import": "./vest/dist/vest.mjs",
+      "require": "./vest/dist/vest.js"
+    },
+    "./superstruct": {
+      "browser": "./superstruct/dist/superstruct.module.js",
+      "umd": "./superstruct/dist/superstruct.umd.js",
+      "import": "./superstruct/dist/superstruct.mjs",
+      "require": "./superstruct/dist/superstruct.js"
+    },
+    "./package.json": "./package.json",
+    "./": "./"
+  },
   "files": [
     "dist",
-    "yup.js",
-    "yup.d.ts",
-    "joi.js",
-    "joi.d.ts",
-    "superstruct.js",
-    "superstruct.d.ts",
-    "zod.js",
-    "zod.d.ts",
-    "vest.js",
-    "vest.d.ts"
+    "yup/package.json",
+    "yup/src",
+    "yup/dist",
+    "zod/package.json",
+    "zod/src",
+    "zod/dist",
+    "vest/package.json",
+    "vest/src",
+    "vest/dist",
+    "joi/package.json",
+    "joi/src",
+    "joi/dist",
+    "superstruct/package.json",
+    "superstruct/src",
+    "superstruct/dist"
   ],
   "publishConfig": {
     "access": "public"
   },
   "scripts": {
-    "clean": "rimraf dist",
-    "prebuild": "yarn clean",
-    "build": "tsc && rollup -c",
-    "lint": "eslint '**/*.{js,ts}'",
-    "lint:fix": "yarn lint -- --fix",
-    "lint:types": "tsc --noEmit",
-    "test": "jest --runInBand",
-    "test:watch": "yarn test --watchAll --coverage"
+    "prepare": "run-s build && check-export-map",
+    "build": "npm-run-all --parallel build:*",
+    "build:src": "microbundle build",
+    "build:zod": "microbundle --cwd zod --globals '@hookform/resolvers=hookformResolvers'",
```

Comment:
That's new to me, `--globals`

Reply:
It specify variable name for external imports in umd/iife bundles